### ISSUE TITLE
static anchors to help/FAQ - link to FAQ entry that opens in the user's language.

### ISSUE THREAD

**Issue by Simon-Laux** [deltachat/deltachat-pages#731]
anchors are generated from headings currently so they are different in every language.
To link from the app to FAQ entries we need a way to link to an entry, regardless what language the user selected it should like to the right place.

Goal is that you can have a link to the FAQ entry and when you click on it it opens in your preferred language.

**Comment by r10s** [comment]
you can get static anchor by adding it in curly brackets after the headline. anchors defined that way overwrite the automatic, language-dependent anchors. they are are also in use already, eg.

[`## Multi-client {#multiclient}`](https://github.com/deltachat/deltachat-pages/blob/master/en/help.md?plain=1#L521C1-L521C31)

defines the anchor `#multiclient` that work on all languages then:

<https://delta.chat/ru/help#multiclient>
<https://delta.chat/zh_CN/help#multiclient>
<https://delta.chat/sq/help#multiclient>
etc.

**Comment by hpk42** [comment]
We also added several anchors to the security section but I guess the main issue is having links that are language neutral and redirect to the lang of the browser that visits the link

**Comment by r10s** [comment]
we have a redirect to from the language-neutral https://delta.chat/help to the user's language (if you click the link, you will be redirected to https://delta.chat/en/help/ or https://delta.chat/de/help/ or whatever)

~~unfortunately, the anchor is lost on that redirect~~ EDIT: this is not true, see below

however, question is  if @Simon-Laux wants to link from the app to the online-help or to the offline-help. for the online-help we would need to fix the redirect, indeed. for the offline help, not sure if there are more issues from deltachat-pages view

**Comment by Simon-Laux** [comment]
Both, mainly website though. but good to know that it can already work for the offline inApp version.

**Comment by Simon-Laux** [comment]
Both, mainly website though. but good to know that it can already work for the offline inApp version.

**Comment by r10s** [comment]
~~thinking it over, not sure if we really can modify  the redirect from `https://delta.chat/help#ANCHOR` to `https://delta.chat/LANG/help#ANCHOR` so that ANCHOR is preserved - iirc, the hash (so the ANCHOR) is not sent to the server~~

~~but we could probably do a `https://delta.chat/help/HASH` that redirects to `https://delta.chat/LANG/help#HASH`?~~

**Comment by r10s** [comment]
according to https://datatracker.ietf.org/doc/html/draft-bos-http-redirect-00 (or probably its successors) it is up to the clients to preserve ANCHOR:

>    If the server returns a response code of 300 ("multiple choice"), 301
   ("moved permanently"), 302 ("moved temporarily") or 303 ("see
   other"), and if the server also returns one or more URIs where the
   resource can be found, then the client SHOULD treat the new URIs as
   if the fragment identifier of the original URI was added at the end.

**and, in fact, all of Firefox, Safari, Chrome do exactly that.**

so, https://delta.chat/help#multiclient already works as expected.

(i assume i was using an outdated hash on my first tries)

**to sum up:**

- static anchors are created as `# some headline {#my-anchor}`

- for online-help, static anchors are addressed as `https://delta.chat/help#my-anchor` 

- for offline-, in-app-help it is up to the app to open the correct language file, from there, the static hash can be used (as the html files are mostly the same)

so, i think, everything's just fine and the issue can be closed.

**Comment by hpk42** [comment]
yes, i agree and nice to know lang-independent FAQ linking is already possible! 
Should use it in blog posts. 
closing this issue -- simon, if there is anything remaining from your original issue, please open a new one and link this one here.
